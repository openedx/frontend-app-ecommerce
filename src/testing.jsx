/* eslint-disable import/no-extraneous-dependencies */
import React from 'react';
import { Provider } from 'react-redux';
import configureMockStore from 'redux-mock-store';
import { IntlProvider } from '@edx/frontend-platform/i18n';

import { render } from '@testing-library/react';
import '@testing-library/jest-dom';

const mockStore = configureMockStore();

jest.mock('@edx/frontend-platform/analytics', () => ({
  sendTrackEvent: jest.fn(),
  sendPageEvent: jest.fn(),
}));

const renderWithProviders = (component, store = null) => render(
  <IntlProvider locale="en">
    {store ? (
      <Provider store={mockStore(store)}>
        {component}
      </Provider>
    ) : (
      component
    )}
  </IntlProvider>,
);

// re-export everything
export * from '@testing-library/react';

// override render method
export { renderWithProviders as render };
