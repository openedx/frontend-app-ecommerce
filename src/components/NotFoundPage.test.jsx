import React from 'react';
import { render } from '../testing';

import NotFoundPage from './NotFoundPage';

describe('<NotFoundPage />', () => {
  describe('Renders NotFoundPage', () => {
    it('renders not found page', () => {
      const { asFragment } = render(<NotFoundPage />);
      expect(asFragment()).toMatchSnapshot();
    });
  });
});
