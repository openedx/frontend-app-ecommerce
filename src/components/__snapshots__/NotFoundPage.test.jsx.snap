// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NotFoundPage /> Renders NotFoundPage renders not found page 1`] = `
<DocumentFragment>
  <div
    class="container-fluid d-flex py-5 justify-content-center align-items-start text-center"
  >
    <p
      class="my-0 py-5 text-muted"
      style="max-width: 32em;"
    >
      The page you're looking for is unavailable or there's an error in the URL. Please check the URL and try again.
    </p>
  </div>
</DocumentFragment>
`;
